packages/tabs/src/styled/StyledTab.ts

```diff
@@ -5,55 +5,120 @@
  * found at http://www.apache.org/licenses/LICENSE-2.0.
  */
 
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
-import classNames from 'classnames';
-import TabStyles from '@zendeskgarden/css-tabs';
-import { retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import styled, { DefaultTheme, css, ThemeProps } from 'styled-components';
+import { retrieveComponentStyles, getColor, DEFAULT_THEME } from '@zendeskgarden/react-theming';
+import stripUnit from 'polished/lib/helpers/stripUnit';
 
 const COMPONENT_ID = 'tabs.tab';
 
 interface IStyledTabProps {
-  isHovered?: boolean;
-  isFocused?: boolean;
-  isActive?: boolean;
   disabled?: boolean;
   isSelected?: boolean;
 }
 
+/**
+ * A high specificity is needed to apply the border-color in vertical orientations
```

Comment:
This comment is way too easy to miss and it doesn't apply to the function. It belongs inline with the line below. We've got to have the ability to comment CSS inline. Styled components docs suggest that it can be stripped on compile: https://www.styled-components.com/docs/tooling#minification. Even if it can't for now (which is a problem that should be fixed) we need to be able to put comments inline and it's worth the added characters to properly document our code.

To that end, I'd like you to recover the comments from https://github.com/zendeskgarden/css-components/tree/master/packages/tabs/src and re-apply them to this migration.

One possible approach that mitigates the added bloat would be to spell out explanations in a header comment, something like this:

```js
/**
 * Consolidated Tab color styling
 *
 * 1. Ensure border-color is applied in vertical orientations.
 * 2. etc.
 * 3. ...
 */
```

And then inline in the CSS-in-JS:

```css
border-color: ${isSelected && 'currentColor'} !important; /* [1] */
```

Reply:
I believe we are minifying correctly at this time. When we initially added styled-component I believe we had the babel plugin misconfigured which is why we were seeing comments original.

I have added in the comments.